front/pages/api/login.ts

```diff
@@ -269,32 +289,34 @@ async function handler(
         },
       });
     }
-
-    if (!m) {
-      m = await createAndLogMembership({
-        workspace: targetWorkspace,
-        userId: user.id,
-        role: "user",
-      });
+    if (membershipInvite.status !== "pending") {
+      membershipInvite = null;
```

Comment:
When can we fall in this case? If the user clicks again on an invite link while they have already subscribed?

Reply:
Good question, the code was already here. Reading the code I would assume that yes, since there is no constraint on the SQL query to find membership invitation. Let me update the SQL query since we have a composite index that covers this query.

Reply:
Done in 2a09867c909711b75f8e9e620a06e07abc730656.